### PR TITLE
Fix DALI installation for python 3.9 version

### PR DESCRIPTION
**Pull request by JanuszL**
- python_requires has the upper python version set to 3.9
  so any 3.9.x falls beyond that check. Rework the check
  to any 3.9.x based version.

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It fixes DALI installation for python 3.9 version

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
    python_requires has the upper python version set to 3.9 so any 3.9.x falls beyond that check. Rework the check to any 3.9.x based version.
 - Affected modules and functionalities:
     python setup.py
 - Key points relevant for the review:
     NA
 - Validation and testing:
     CI
 - Documentation (including examples):
     NA

**JIRA TASK**: *[NA]*

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

**Review by jantonguirao — APPROVED** [review]

**Review by mzient — APPROVED** [review]

**Review by klecki — APPROVED** [review]

File: dali/python/setup.py.in [hunk 1]
```diff
@@ -23,7 +23,7 @@ setup(name='nvidia-dali@DALI_FLAVOR_MINUS@-cuda@CUDA_VERSION_SHORT_DIGIT_ONLY@',
       packages=find_packages(),
       include_package_data=True,
       zip_safe=False,
-      python_requires='>=3.6, <=3.9',
+      python_requires='>=3.6, <=3.9.99',
```
Comment (klecki) [inline dali/python/setup.py.in:26]: Hmm, actually I would suggest, either:
```suggestion
      python_requires='>=3.6, <3.10',
```
or
```suggestion
      python_requires='>=3.6, <3.9.*',
```
As both _should_ work:
https://packaging.python.org/guides/distributing-packages-using-setuptools/#python-requires
https://www.python.org/dev/peps/pep-0440/#version-specifiers

**Review by klecki — CHANGES_REQUESTED** [review]

File: dali/python/setup.py.in (hunk 1 above)
Comment (klecki) [reply dali/python/setup.py.in:26]: Using arbitrary .99 just looks weird.

Comment (JanuszL) [reply dali/python/setup.py.in:26]: Done

**Comment by JanuszL** [comment]
!build

**Review by klecki — APPROVED** [review]

**Pull request merged**